src/CalendarComponent.jsx

```diff
@@ -22,18 +21,25 @@ class CalendarComponent extends React.Component {
 	* @description init the js date flatpickr component
 	*/
 	componentDidMount() {
-		const options = {
-			onChange: this.onChange,
-			onOpen: this.onOpen,
-			onClose: this.onClose,
-			altInput: true,
-			allowInput: true,
-			altFormat: 'D M d, Y', // TODO localize
-			defaultDate: this.props.value
-		};
+		// flatpickr uses `window` on import,
+		// which breaks on server-sider render.
+		// lazy-loading flatpickr ensures it is
+		// imported only in clientside envs.
+		const Flatpickr = require('flatpickr');
```

Comment:
`require()` is an interesting option - this will technically include flatpickr in both the server-side and client-side bundles, but it will only be actually imported in the client. It also uses webpack-specific syntax (since `require` isn't a native feature of browser JS). The alternative, using `import('flatpickr').then()` would actually make an async HTTP request for the module on the client, which means it's slower, but would also technically be compliant with [this TC39 proposal](https://github.com/tc39/proposal-dynamic-import). No need to change, just something to be aware of.